integration-libs/cdc/root/guards/cdc-user.guard.ts

```diff
@@ -0,0 +1,44 @@
+/*
+ * SPDX-FileCopyrightText: 2023 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Injectable } from '@angular/core';
+import { CanActivate, Router, UrlTree } from '@angular/router';
+import {
+  GlobalMessageService,
+  GlobalMessageType,
+  SemanticPathService,
+} from '@spartacus/core';
+import { UserGuard } from '@spartacus/organization/administration/core';
+import { CdcB2BUserService } from '../../organization/administration/public_api';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class CdcUserGuard extends UserGuard implements CanActivate {
+  constructor(
+    protected globalMessageService: GlobalMessageService,
+    protected cdcb2bUserService: CdcB2BUserService,
+    protected semanticPathService: SemanticPathService,
+    protected router: Router
+  ) {
+    super(globalMessageService, cdcb2bUserService, semanticPathService, router);
+  }
+
+  canActivate(): boolean | UrlTree {
```

Comment:
I just realized injecting CdcB2BUserService into this guard is creating a circular dependency within cdc modules. hence i have changed the implementation. Can you check it again

Reply:
This guard fires only on administration/organization pages (b2b). Personally, I am not convenienced if we should keep this guard responsible for such logic outside `@spartacus/cdc/organization/administration` sub-library.

As you mentioned it also produces some potential circular dependencies between libs.

My idea is to move this guard into `organization/administration/guards` folder and then try to overwrite the config 
 in `cdc-administration.module.ts` to use `CdcUserGuard` instead of `UserGuard`.

```
 provideDefaultConfig(<CmsConfig>{
      cmsComponents: {
        ManageUsersListComponent: {
          providers: [
            {
              provide: UserListService,
              useExisting: CdcUserListService,
            },
            userCmsConfig.cmsComponents?.ManageUsersListComponent?.providers ||
              [],
          ],
          childRoutes: {
            children: [
              {
                path: 'create',
                component: UserFormComponent,
                canActivate: [CdcUserGuard],
              },
              {
                path: `:${ROUTE_PARAMS.userCode}`,
                component: UserDetailsComponent,
                data: {
                  cxPageMeta: { breadcrumb: 'orgUser.breadcrumbs.details' },
                },
                children: [
                  {
                    path: `edit`,
                    component: UserFormComponent,
                    canActivate: [CdcUserGuard],
                  },
                ],
              },
            ],
          },
        },
      },
    }),
```

By this `root` still will be responsible for stuff not related with any other sub-library.

Reply:
But also I am open for other ideas @anjana-bl

Reply:
Hi Mateauz, thanks for the sample code. But if we do like this, then we have to copy the entire content of user.config.ts . Else, the remaining navigations wont work, right? eg: we cannot navigate to 'approvers' from user details screen. Is it okay if I copy the entire content of user.config.ts into a new file, say, cdc-user.config.ts and make necessary changes (eg: replace UserGuard with CdcUserGuard) and use it?